### PR TITLE
feat(operators): Adding 'overlaps' operator support

### PR DESCRIPTION
**Pull request by arnaud16571542**
This PR add support to the 'overlaps' postgresql extended operator.
Currently the 'contains' operator return rows which contains all items in submitted array.
The new 'overlaps' operator return rows which contains any items in submitted array.

The 'overlaps' operator use the && postgresql array operator, and is described by the postgresql doc as "Do the arrays overlap, that is, have any elements in common?"
https://www.postgresql.org/docs/current/functions-array.html

    Code is updated.
    Documentation is updated.
    Tests are updated & OK.

## Checklist

- [X] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [X] `npm test` passes on your machine
- [X] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [X] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by achrinza** [comment]
Thanks for the PR. Please consider updating the existing PR instead of opening new ones so that it's easier for the maintainers to follow and track. Thanks!

File: README.md [hunk 1]
```diff
@@ -564,7 +565,7 @@ rows where the stored value contains all of the items specified by the query.
 The operator is implemented using PostgreSQL [array operator
 `@>`](https://www.postgresql.org/docs/current/functions-array.html).
 
-**Note** The fields you are querying must be setup to use the postgresql array data type - see [Defining models](#defining-models) above.
+**Note** The fields you are querying must be setup to use the postgresql array data type - see [Defining models](#defining-models) above.
```
Comment (dhmlau) [inline README.md:568]: could you please remove the trailing space?

File: lib/postgresql.js [hunk 2]
```diff
@@ -547,6 +547,9 @@ PostgreSQL.prototype.buildExpression = function(columnName, operator,
     case 'contains':
       return new ParameterizedSQL(columnName + ' @> array[' + operatorValue.map((v) => `'${v}'`) + ']::'
         + propertyDefinition.postgresql.dataType);
+    case 'overlaps':
```
Comment (dhmlau) [inline lib/postgresql.js:550]: Code linter complained about "Trailing spaces not allowed".

File: test/postgresql.test.js [hunk 3]
```diff
@@ -281,6 +281,24 @@ describe('postgresql connector', function() {
     found.map(p => p.title).should.deepEqual(['Growing LoopBack Community']);
   });
 
+  it('should support the "overlaps" where filter for array type field', async () => {
```
Comment (dhmlau) [inline test/postgresql.test.js:284]: same as above

File: lib/postgresql.js [hunk 4]
```diff
@@ -547,6 +547,9 @@ PostgreSQL.prototype.buildExpression = function(columnName, operator,
     case 'contains':
       return new ParameterizedSQL(columnName + ' @> array[' + operatorValue.map((v) => `'${v}'`) + ']::'
         + propertyDefinition.postgresql.dataType);
+    case 'overlaps': 
+      return new ParameterizedSQL(columnName + ' && array[' + operatorValue.map((v) => `'${v}'`) + ']::'
+        + propertyDefinition.postgresql.dataType);
```
Comment (mgabeler-lee-6rs) [inline lib/postgresql.js:552]: SQL injection here, see https://github.com/loopbackio/loopback-connector-postgresql/security/advisories/GHSA-j259-6c58-9m58

**Comment by samarpanB** [comment]
@arnaud16571542 are you still working on this PR ? If not, can we create a new PR with all the fixes and changes ? This is an important feature for the community.

**Pull request closed without merging**